app/src/util/env_config.py

```diff
@@ -1,16 +1,30 @@
-import os
+import logging
 
-from pydantic import BaseSettings
+from pydantic import BaseModel
 
-import src
+logger = logging.getLogger(__name__)
 
-env_file = os.path.join(
-    os.path.dirname(os.path.dirname(src.__file__)),
-    "config",
-    "%s.env" % os.getenv("ENVIRONMENT", "local"),
-)
 
+class PydanticBaseEnvConfig(BaseModel):
+    """Base class for application configuration.
+
+    Similar to Pydantic's BaseSettings class, but we implement our own method to override from the
+    environment so that it can be run later, after an instance was constructed."""
 
-class PydanticBaseEnvConfig(BaseSettings):
     class Config:
-        env_file = env_file
+        validate_assignment = True
```

Comment:
Could require a prefix option here too to provide some namespacing for the various configs, e.g., have the PostgresConfig environment variables all start with a `PG_DB_` prefix.

Reply:
(The implementation on PFML does this)